ev3dev/__init__.py

```diff
@@ -174,23 +177,24 @@ def get_index(file):
         if name_exact:
             self._path = classpath + '/' + name_pattern
             self._device_index = get_index(name_pattern)
-            self.connected = True
         else:
             try:
                 name = next(list_device_names(classpath, name_pattern, **kwargs))
                 self._path = classpath + '/' + name
                 self._device_index = get_index(name)
-                self.connected = True
             except StopIteration:
                 self._path = None
                 self._device_index = None
-                self.connected = False
+                raise DeviceNotFoundException("No device matching the given requirements was found.")
```

Comment:
`"%s is not connected" % self` would let the user know exactly which device isn't connected